### PR TITLE
SC WS: handling of broken init contract code

### PR DESCRIPTION
**Pull request by velzevur**
PT [SC WS: Broken init call to a contract crashes the FSM](https://www.pivotaltracker.com/story/show/164243431)

### CODE REVIEW

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 1]
```diff
@@ -5901,7 +5903,9 @@ lift_reason(#{ <<"message">> := <<"Rejected">>
     Codes = lists:sort([Code || #{<<"code">> := Code} <- Data]),
     E#{ <<"reason">> => data_code_to_reason(Codes) };
 lift_reason(#{ <<"code">> := Code } = E) ->
-    E#{ <<"reason">> => code_to_reason(Code) }.
+    E#{ <<"reason">> => code_to_reason(Code) };
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: Not `data_code_to_reason([Code])`?

**Comment by velzevur** [comment]
I will remove the hardcoded contract and will replace it with a proper _broken_ code instead.
Adding a `WIP`

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 2]
```diff
@@ -5911,12 +5915,10 @@ data_code_to_reason([1003])      -> <<"invalid_pubkeys">>;
 data_code_to_reason([1004])      -> <<"call_not_found">>;
 data_code_to_reason([1005])      -> <<"broken_encoding: accounts">>;
 data_code_to_reason([1006])      -> <<"broken_encoding: contracts">>;
+data_code_to_reason([1007])      -> <<"contract_init_failed">>;
```
Comment (lucafavatella) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: Is this meant to integrate https://github.com/aeternity/protocol/blame/3a00b538ed47efa6ab7bb205f07456e06134ff44/node/api/channel_ws_api_json-rpc.md#L179 ?

Comment (velzevur) [reply apps/aehttp/test/aehttp_integration_SUITE.erl]: Addressed in aeternity/protocol#320

**Pull request merged**